src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventSource.cs

```diff
@@ -3799,6 +3805,159 @@ private bool SelfDescribingEvents
             }
         }
 
+#if NATIVEAOT
+        // If EventSource feature is enabled, default EventSources need to be initialized for NativeAOT
+        // In CoreCLR, this is done via a call from the runtime as part of coreclr_initialize
+#pragma warning disable CA2255
+        [ModuleInitializer]
+#pragma warning restore CA2255
+#endif
+        internal static void InitializeDefaultEventSources()
+        {
+            if(!EventSource.IsSupported)
+            {
+                return;
+            }
+
+// NOTE: this define is being used inconsistently. Most places mean just EventPipe support, but then a few places use
+// it to mean other aspects of tracing such as these EventSources.
+#if FEATURE_PERFTRACING
+            _ = NativeRuntimeEventSource.Log;
+            _ = RuntimeEventSource.Log;
```

Comment:
I assume these doesn't  need GC.KeepAlive anymore. right?

Reply:
thats correct, the static field roots them.